app/src/main/java/com/blockeq/stellarwallet/activities/RecoverWalletActivity.kt

```diff
@@ -107,6 +138,41 @@ class RecoverWalletActivity : BaseActivity() {
     //endregion
 
     //region Helper functions
+    private fun highlightMnemonic() {
+        val wordListBIP39 = WordList.ENGLISH.words.toHashSet()
+
+        val tokens = phraseEditText.text.split(" ".toRegex()).dropLastWhile { it.isEmpty() }
+        var startIndex = 0
+        var endIndex = 0 //hi h
+
+        for (word in tokens) {
+
+            // Color the last word
+            endIndex += word.length
+
+            val colorText = if (!wordListBIP39.contains(word)) {
+                ForegroundColorSpan(Color.RED)
+            } else {
+                ForegroundColorSpan(Color.BLACK)
+            }
+
+            phraseEditText.text.setSpan(colorText, startIndex, endIndex, Spannable.SPAN_INCLUSIVE_EXCLUSIVE)
+            startIndex += word.length + 1
+            ++endIndex
+        }
+    }
+
+    private fun highlightSeed() {
+        val seedText = secretKeyEditText.text
+        val colorText = if (seedText.length != Constants.STELLAR_ADDRESS_LENGTH || seedText[0] != 'S') {
+            ForegroundColorSpan(Color.RED)
```

Comment:
same as before reuse the default color for BLACK, and move the constant Color.RED to the top.